privaterelay/templatetags/relay_tags.py

```diff
@@ -51,3 +52,10 @@ def premium_plan_price(accept_lang, cc=None):
 def premium_subscribe_url(accept_lang=None, cc=None):
     plan_id = premium_plan_id(accept_lang, cc)
     return f'{settings.FXA_SUBSCRIPTIONS_URL}/products/{settings.PREMIUM_PROD_ID}?plan={plan_id}'
+
+
+@register.filter
+def full_address(alias, user_profile):
+    if type(alias) == RelayAddress:
+        return '%s@%s' % (alias.address, alias.domain_value)
+    return '%s@%s.%s' % (alias.address, user_profile.subdomain, alias.domain_value)
```

Comment:
I was hoping we don’t use full_address  template tag  because I wanted to prevent the formatting of the strings to be done in multiple different function. Wondering if we can use the `cached_property` for the `full_address` property too?

Reply:
Putting `cached_property` on `Domain.full_address` didn't work when I tried it. 😢 Near as I can tell (though I didn't debug thru the Django code itself), calling a `cached_property` from different points in a template passes a different `self` argument value to the decorator, so it doesn't return from the cache - it makes the full DB query each time.